packages/kbn-performance-testing-dataset-extractor/src/es_client.ts

```diff
@@ -88,81 +100,92 @@ const addRangeFilter = (range: { startTime: string; endTime: string }): QueryDsl
   };
 };
 
-export function initClient(options: ClientOptions, log: ToolingLog) {
-  const client = new Client({
-    node: options.node,
-    auth: {
-      username: options.username,
-      password: options.password,
-    },
-  });
+export class ESClient {
+  client: Client;
+  log: ToolingLog;
 
-  return {
-    async getKibanaServerTransactions(
-      buildId: string,
-      journeyName: string,
-      range?: { startTime: string; endTime: string }
-    ) {
-      const filters = [
-        { field: 'transaction.type', value: 'request' },
-        { field: 'processor.event', value: 'transaction' },
-        { field: 'labels.testBuildId', value: buildId },
-        { field: 'labels.journeyName', value: journeyName },
-      ];
-      const queryFilters = filters.map((filter) => addBooleanFilter(filter));
-      if (range) {
-        queryFilters.push(addRangeFilter(range));
-      }
-      return await this.getTransactions(queryFilters);
-    },
-    async getFtrTransactions(buildId: string, journeyName: string) {
-      const filters = [
-        { field: 'service.name', value: 'functional test runner' },
-        { field: 'processor.event', value: 'transaction' },
-        { field: 'labels.testBuildId', value: buildId },
-        { field: 'labels.journeyName', value: journeyName },
-        { field: 'labels.performancePhase', value: 'TEST' },
-      ];
-      const queryFilters = filters.map((filter) => addBooleanFilter(filter));
-      return await this.getTransactions(queryFilters);
-    },
+  constructor(options: ClientOptions, log: ToolingLog) {
+    this.client = new Client({
+      node: options.node,
+      auth: {
+        username: options.username,
+        password: options.password,
+      },
+    });
+    this.log = log;
+  }
 
-    async getTransactions(queryFilters: QueryDslQueryContainer[]) {
-      const searchRequest: SearchRequest = {
-        body: {
-          track_total_hits: true,
-          sort: [
-            {
-              '@timestamp': {
-                order: 'asc',
-                unmapped_type: 'boolean',
-              },
+  async getTransactions<T>(queryFilters: QueryDslQueryContainer[]) {
+    const searchRequest: SearchRequest = {
+      body: {
+        track_total_hits: true,
```

Comment:
Is there a reason why we track total hits? As far as I could see we only return the hits themselves but never check the number of hits? I'm asking because tracking hits comes at a cost. As it's implemented now there won't be a practical difference because [total hits are tracked accurately for up to 10.000 results](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-your-data.html#track-total-hits) but I don't see a reason why we need to enforce this here.

Reply:
You are right, I can't recall why we initially decided to have. Good to know it is affecting the query performance, removing.